projects/control-service/projects/pipelines_control_service/src/main/java/com/vmware/taurus/service/deploy/DeploymentService.java

```diff
@@ -193,6 +199,21 @@ private void saveDeployment(DataJob dataJob, JobDeployment jobDeployment) {
     }
   }
 
+  /**
+   * As pythonVersion is optional, we need to check if it is passed. And if it is, we need to
+   * validate that the python version is supported by the Control Service. If it is not, we need to
+   * fail the operation, as we don't have sufficient information for the user's intent to deploy the
+   * data job.
+   *
+   * @param pythonVersion The python version to be used for the data job deployment.
+   */
+  public void validatePythonVersionIsSupported(String pythonVersion) {
+    if (pythonVersion != null && !supportedPythonVersions.isPythonVersionSupported(pythonVersion)) {
+      throw new UnsupportedPythonVersionException(
+          pythonVersion, supportedPythonVersions.getSupportedPythonVersions().toString());
```

Comment:
Better `supportedPythonVersions.getSupportedPythonVersions().toString()` move that to the UnsupportedPythonVersionException and pass the list.

Reply:
Good point, fixed. Thanks